Tasks/GooglePlayStatusUpdate/google-play-status-update.ts

```diff
@@ -0,0 +1,88 @@
+import * as path from 'path';
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as googleutil from './googleutil';
+
+async function run() {
+    try {
+        tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+        const authType: string = tl.getInput('authType', true);
+        let key: googleutil.ClientKey = {};
+        if (authType === 'JsonFile') {
+            const serviceAccountKeyFile: string = tl.getPathInput('serviceAccountKey', false);
+            if (!serviceAccountKeyFile) {
+                throw new Error(tl.loc('JsonKeyFileNotFound'));
+            }
+            const stats: tl.FsStats = tl.stats(serviceAccountKeyFile);
+            if (stats && stats.isFile()) {
+                key = require(serviceAccountKeyFile);
+            } else {
+                console.error(tl.loc('InvalidAuthFile'));
+                throw new Error(tl.loc('InvalidAuthFilewithName', serviceAccountKeyFile));
+            }
+        } else if (authType === 'ServiceEndpoint') {
+            const serviceEndpoint: tl.EndpointAuthorization = tl.getEndpointAuthorization(tl.getInput('serviceEndpoint', true), true);
+            if (!serviceEndpoint) {
+                throw new Error(tl.loc('EndpointNotFound'));
+            }
+            key.client_email = serviceEndpoint.parameters['username'];
+            key.private_key = serviceEndpoint.parameters['password'].replace(/\\n/g, '\n');
+        }
+
+        const packageName: string = tl.getPathInput('packageName', true);
+        const status: string = tl.getInput('status', true);
+        const userFrac: string = tl.getInput('userFraction', false);
+        let userFraction: number = Number.NaN;
+        const trackName: string = tl.getInput('track', true);
+
+        // Constants
+        const globalParams: googleutil.GlobalParams = { auth: null, params: {} };
+
+        const jwtClient = googleutil.getJWT(key);
+        const edits: any = googleutil.publisher.edits;
+
+        globalParams.auth = jwtClient;
+        googleutil.updateGlobalParams(globalParams, 'packageName', packageName);
+
+        console.log(tl.loc('Authenticating'));
+        await jwtClient.authorize();
+        const edit = await googleutil.getNewEdit(edits, globalParams, packageName);
+        googleutil.updateGlobalParams(globalParams, 'editId', edit.id);
+
+        console.log(tl.loc('GetTrackInfo', trackName));
+        const track: googleutil.Track = await googleutil.getTrack(edits, packageName, trackName);
+        tl.debug('Track: ' + JSON.stringify(track));
+
+        if (track.releases.length <= 0) {
+            throw new Error(tl.loc('EmptyReleases'));
+        }
+        const firstRelease = track.releases[0];
+
+        if (typeof(userFrac) === 'undefined' || userFrac == null || userFrac.replace(/(^s*)|(s*$)/g, '').length === 0) {
```

Comment:
`userFrac` may either be undefined or string, so null check is redundant

```suggestion
        if (userFrac === undefined || userFrac.replace(/(^s*)|(s*$)/g, '').length === 0) {
```